### DISCUSSION TITLE
Whats the best approach to run multiple tests on Github Actions or Jenkins

### DISCUSSION THREAD

**Discussion by Thilaknath** [artilleryio/artillery#2288] [Artillery]
Hello Team, 
We would possibly have multiple test scenarios under the directory tests/performance/ Is it possible for artillery to execute all tests in a given folder sequentially ?

**Comment by bernardobridge** [comment]
Hi @Thilaknath 👋,

It's not currently possible for Artillery to execute all tests in a folder sequentially, although we're certainly interested in knowing more use-cases, as it helps us know what to build 👍 .

---

You can typically orchestrate that at the level of your CI (Jenkins/GH Actions/other), with either a custom script, or multiple jobs. For example, in a previous company, we had a custom `generic` Jenkins job that was capable of running any of the load tests in the repo. Then, we would have `orchestrator` jobs that could run multiple of those (either in parallel or sequentially), for different reasons. 

For instance, we would run every load test in the repo nightly with a smaller RPS, to gather baselines - we had the `orchestrator` job that had the list of load tests, the desired environment, and set a few more things, as well as the cron. That job would then call the `generic` job N times (where N is the number of tests), sequentially. This is a fairly easy pattern to implement and scales well, as then your on-demand job can be used for multiple things.

With our official [Github Action](https://www.artillery.io/docs/cicd/github-actions), you can have a very similar pattern by using multiple workflow files. But it's a pattern that should work in most CI platforms.

_Note: If you're using the Artillery Cloud Dashboard, then you can even tag your tests with certain things to identify they were run together, like a unique ID per orchestration job run, owner (if a certain team owns it), etc. This makes it quite easy to filter results after the fact._

---

Hope that helps!

**Reply by hassy** [reply]
Great answer @bernardobridge! If all you want is to run all tests in the same job @Thilaknath then something like this should work:

```sh
for fn in $(ls ./tests/performance/*.yml); do
  artillery run $fn
done
```

You can tweak the args to `artillery` as needed.
